lib/trino-parquet/src/main/java/io/trino/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -218,6 +228,47 @@ public boolean matches(Map<ColumnDescriptor, Long> valueCounts, ColumnIndexStore
         return true;
     }
 
+    /**
+     * Should the Parquet Reader process a file section with the specified bloomfilter Store
+     *
+     * @param bloomFilterStore bloomfilter Store
+     */
+    public boolean matches(BloomFilterStore bloomFilterStore)
+    {
+        requireNonNull(bloomFilterStore, "bloomFilterStore is null");
+
+        if (effectivePredicate.isNone()) {
+            return false;
+        }
+        Map<ColumnDescriptor, Domain> effectivePredicateDomains = effectivePredicate.getDomains()
+                .orElseThrow(() -> new IllegalStateException("Effective predicate other than none should have domains"));
+
+        for (ColumnDescriptor column : columns) {
+            Domain effectivePredicateDomain = effectivePredicateDomains.get(column);
+
+            // the bloom filter bitset contains only non-null values so isn't helpful
```

Comment:
Could you add in `TestBloomFilterStore` class also test cases showcasing the specific cases:

- null values 
- domainCompactionThreshold exceeded

where the parquet predicate doesn't actually match the bloom filter store, but nevertheless the `matches(bloomFilterStore)` operation result is `true`. 


e.g. : matching against a domain greater than the `DOMAIN_COMPACTION_THRESHOLD` that doesn't overlap the bloom filter store:

```
            BloomFilterStore bloomFilterStore = generateBloomFilterStore(tempFile, true, Arrays.asList(60, 61, 62, 63, 64, 65), javaIntObjectInspector);

            TupleDomain<ColumnDescriptor> domain = TupleDomain.withColumnDomains(singletonMap(columnDescriptor, Domain.create(SortedRangeSet.copyOf(INTEGER,
                    ImmutableList.of(Range.range(INTEGER, 200L, true, 300L, true))), false)));
            TupleDomainParquetPredicate parquetPredicate = new TupleDomainParquetPredicate(domain, singletonList(columnDescriptor), UTC, DOMAIN_COMPACTION_THRESHOLD);
            // bloomfilter store has the column, and ranges overlap
            assertTrue(parquetPredicate.matches(bloomFilterStore));
```

Reply:
Hi, the 
domainCompactionThreshold exceeded case is now covered in testMatchesWithBloomFilterExpand

Reply:
I also added two tests for null